src/quantization/computer.h

```diff
@@ -0,0 +1,57 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+
+#include "metric_type.h"
+
+namespace vsag {
+using DataType = float;
+
+template <typename T>
+class Quantizer;
+
+class ComputerInterface {
+protected:
+    ComputerInterface() = default;
+};
+
+template <typename T>
+class Computer : public ComputerInterface {
+public:
+    ~Computer() {
+        delete[] buf_;
+    }
+
+    explicit Computer(const T& quantizer) : quantizer_(&quantizer){};
```

Comment:
There can replace to `const T*` ?
The const reference is not safe, example:

```C++
#include <iostream>

const int* ptr = nullptr;

void test(const int& ref) {
    ptr = &ref;
    std::cout << *ptr << std::endl;
}

void test2() {
    int num = 2;
    test(1);
    test(num);
}

int main(int argc, char* argv[]) {

    test2();
    std::cout << *ptr << std::endl; // access undefined memory

    return 0;
}
```

Reply:
fixed, use const T* instead